### PR TITLE
Epic: 8497 - Nice to have feature - Reorder an existing request

### PR DESCRIPTION
**Pull request by labkey-bpatel**
#### Rationale
Add a “Reorder” button/link to an existing or completed purchase request so that the users do not have to type in all the detailed information of a new order in the case they want to duplicate the same order.
Spec [here](https://docs.google.com/document/d/1GQjSlRykqD0KDkMY6FafuGFxr608bT3APs7TmMvvOzA/edit#).

#### Changes
* Add Reorder link to requester and admin views. 
* Update data entry page to add a 'Reorder' button. 
* Reset Accounts to charge, Attachments, and Purchase Admin panel.
* Handle save for a reorder request.

### CODE REVIEW

File: WNPRC_Purchasing/src/client/RequestEntry/RequestEntry.tsx [hunk 1]
```diff
@@ -100,14 +118,14 @@ export const App: FC = memo(() => {
...
-                                qcState: requestOrderVals[0].qcState,
-                                program: requestOrderVals[0].program,
-                                confirmationNum: requestOrderVals[0].confirmationNum,
-                                invoiceNum: requestOrderVals[0].invoiceNum,
-                                orderDate: requestOrderVals[0].orderDate ? new Date(requestOrderVals[0].orderDate) : null,
-                                cardPostDate: requestOrderVals[0].cardPostDate ? new Date(requestOrderVals[0].cardPostDate) : null,
+                                assignedTo: isReorder ? "" : requestOrderVals[0].assignedTo,
+                                paymentOption: isReorder ? "" : requestOrderVals[0].paymentOptionId,
+                                qcState: isReorder ? pendingState : requestOrderVals[0].qcState,
+                                program: isReorder ? "4" : requestOrderVals[0].program,
```
Comment (labkey-jeckels) [inline WNPRC_Purchasing/src/client/RequestEntry/RequestEntry.tsx]: Can this be a constant? I assume that we have a similar default used when we create brand new requests?

File: WNPRC_Purchasing/src/client/components/LineItemRow.tsx [hunk 2]
```diff
@@ -59,22 +60,22 @@ export const LineItemRow: FC<LineItemProps> = memo(props => {
                             value={model.item}
                             hasError={model.errors?.find(field => field.fieldName === 'item')}
                             onChange={onValueChange}
-                            isReadOnly={hasRequestId && (isRequester || isReceiver)}
+                            isReadOnly={hasRequestId && ((isRequester && !isReorder) || isReceiver)}
```
Comment (labkey-jeckels) [inline WNPRC_Purchasing/src/client/components/LineItemRow.tsx]: This somewhat complex expression is repeated a bunch of times. Can we pull it into a single variable and use that each time?

File: WNPRC_Purchasing/resources/queries/ehr_purchasing/purchasingRequestsOverview.query.xml [hunk 3]
```diff
@@ -35,6 +35,10 @@
                     </column>
                     <column columnName="OtherAcctAndInves"/>
                     <column columnName="requester"/>
+                    <column columnName="orderDate"/>
+                    <column columnName="reorderAction">
+                        <required>false</required>
```
Comment (labkey-jeckels) [inline WNPRC_Purchasing/resources/queries/ehr_purchasing/purchasingRequestsOverview.query.xml]: Do you need both this and the setRequired(false) in WNPRC_PurchasingCustomizer.java?

**Review by labkey-jeckels — APPROVED** [review]
Small suggestions. No need for additional review.

**Comment by labkey-bpatel** [comment]
All tests green woohoo! merging

**Pull request merged**